### PR TITLE
Replace bind by arrow functions (PasswordLogin.js)

### PR DESCRIPTION
**Pull request by czeidler**
The member functions `onUsernameFocus` was not bound which caused
uncaught exceptions. Use arrow functions instead to make it unnecessary
to bind member functions.

Signed-off-by: Clemens Zeidler <clemens.zeidler@gmail.com>

### CODE REVIEW

File: src/components/views/auth/PasswordLogin.tsx [hunk 1]
```diff
@@ -26,10 +26,41 @@ import {ValidatedServerConfig} from "../../../utils/AutoDiscoveryUtils";
...
+    phonePrefix?: string;
+    loginType: string;
+}
+
 /**
  * A pure UI component which displays a username/password form.
  */
-export default class PasswordLogin extends React.Component {
+export default class PasswordLogin extends React.Component<IProps, IState> {
     static propTypes = {
```
Comment (turt2live) [inline src/components/views/auth/PasswordLogin.tsx:64]: We drop `propTypes` when we do the conversion.

File: src/components/views/auth/PasswordLogin.tsx [hunk 2]
```diff
@@ -93,7 +124,7 @@ export default class PasswordLogin extends React.Component {
         let username = ''; // XXX: Synapse breaks if you send null here:
         let phoneCountry = null;
         let phoneNumber = null;
```
Comment (turt2live) [inline src/components/views/auth/PasswordLogin.tsx:126]: These should get types too.

File: src/components/views/auth/PasswordLogin.tsx [hunk 3]
```diff
@@ -135,7 +166,7 @@ export default class PasswordLogin extends React.Component {
         );
     }
 
-    onUsernameChanged = (ev) => {
+    onUsernameChanged = (ev: any) => {
```
Comment (turt2live) [inline src/components/views/auth/PasswordLogin.tsx:169]: `any` is not acceptable during a conversion - it needs to be a defined type, or a comment giving a good reason why the type can't be defined.

**Review by turt2live — CHANGES_REQUESTED** [review]
The first commit looks fine overall - the TypeScript conversion is incomplete. There are types missing on most functions (many of which github won't let me leave inline comments on), and the imports haven't been fixed as part of the migration (we remove instances of `getComponent()` when we do a conversion).

**Comment by czeidler** [comment]
There is only one implicit any left (as far as I see):

`let forgotPasswordJsx;`

but I think that is ok for this case?

Do we allow implicit any?

**Comment by czeidler** [comment]
@turt2live I applied the requested changes

**Comment by jryans** [comment]
It looks like @t3chguy happened to make a similar conversion in https://github.com/matrix-org/matrix-react-sdk/pull/5433 which has been merged, so this one appears to no longer be needed. Thanks for the contribution in any case, and please do look around for other areas to help out. 😄

**Pull request closed without merging**